src/pages/dataTableUtilities/actions/cellActions.js

```diff
@@ -36,6 +36,30 @@ export const refreshIndicatorRow = route => ({
   payload: { route },
 });
 
+export const editLocationDescription = (newValue, rowKey, route) => (dispatch, getState) => {
+  const { data, keyExtractor } = selectPageState(getState());
+
+  const objectToEdit = data.find(row => keyExtractor(row) === rowKey);
+
+  if (objectToEdit) {
+    UIDatabase.write(() =>
+      UIDatabase.update('Location', { ...objectToEdit, description: newValue })
+    );
+    dispatch(refreshRow(rowKey, route));
+  }
+};
+
+export const editLocationCode = (newValue, rowKey, route) => (dispatch, getState) => {
+  const { data, keyExtractor } = selectPageState(getState());
+
+  const objectToEdit = data.find(row => keyExtractor(row) === rowKey);
+
+  if (objectToEdit) {
+    UIDatabase.write(() => UIDatabase.update('Location', { ...objectToEdit, code: newValue }));
+    dispatch(refreshRow(rowKey, route));
+  }
+};
```

Comment:
I find these interesting. A lot you could do to make this more dry. Extra prop for the field? Also a prop for the data type? Then wrap it in another thunk ?

🤷

Reply:
Or could pass an object with key constants? That way could update multiple values (if ever needed) and lookup the type in the reducer using the key. Also keeps the number of parameters at the same length, e.g `(newValues, rowKey, route)`.

Reply:
yeah a lot you can do - just not sure any of it is worth making the code more confusing/convoluted by making it more generic - does it really bring any benefit, I don't know if it does!